FreeRTOS/Source/tasks.c

```diff
@@ -2789,12 +2936,24 @@ BaseType_t xSwitchRequired = pdFALSE;
 						mtCOVERAGE_TEST_MARKER();
 					}
 
-					/* Place the unblocked task into the appropriate ready
-					list. */
+#if ( configUSE_EDF_SCHEDULER == 1 )
+                    // h_edf_10. calculate new deadline then add task to EDF Ready list
+                    listSET_LIST_ITEM_VALUE(&((pxTCB)->xStateListItem) , ((TickType_t)(((pxTCB)->xTaskPeriod) + xTickCount)) );
+                    // also updates idle task deadline
+                    listSET_LIST_ITEM_VALUE( &( ( xIdleTaskHandle )->xStateListItem ), ( xIdleTaskHandle)->xTaskPeriod + xTickCount);
+#endif
+
+                    /* Place the unblocked task into the appropriate ready
+                    list. */
 					prvAddTaskToReadyList( pxTCB );
 
-					/* A task being unblocked cannot cause an immediate
-					context switch if preemption is turned off. */
+#if( configUSE_EDF_SCHEDULER == 1 )
+                    // h_edf_11. always switch context to meet EDF criteria
```

Comment:
Here you have to make a condition that the task with the sooner deadline must preempt the task with a larger deadline

Note: -Deadline is stored in a variable named "xItemValue"